prefq/examples/updated_imitation_modules.py

```diff
@@ -0,0 +1,232 @@
+"""Adjust imitation modules required for PrefQ to the new gymnasium API."""
+
+# These functions are taken from [1][2] for integration into the new gymnasium API.
+# [1] https://imitation.readthedocs.io/en/latest/_modules/imitation/util/util.html
+# [2] https://github.com/HumanCompatibleAI/imitation/pull/716/commits/63e9112faf92d195b4b80169d213c3ad17c49c2e
+
+# Disable formatting and linters so we can keep the example as close to the original as possible for now.
+# fmt: off
+# pylint: skip-file
+
+import functools
+import os
+import shutil
+import tempfile
+import uuid
+from typing import Any, Callable, List, Mapping, Optional, Sequence, Union, overload
+
+import cv2
+import gymnasium
+import numpy as np
+from stable_baselines3.common import monitor
+from stable_baselines3.common.vec_env import DummyVecEnv, SubprocVecEnv, VecEnv
+
+# This Class has been taken from
+# https://github.com/HumanCompatibleAI/imitation/pull/716/commits/63e9112faf92d195b4b80169d213c3ad17c49c2e
+
+
+# Due to the update of the gymnasium API, several changes had to be made,
+# in order to make the code compatible with the new package
+# https://stackoverflow.com/questions/73195438/openai-gyms-env-step-what-are-the-values
```

Comment:
I'm a bit confused why we need to make these changes, since the async prefs PR should be using gymnasium by now. Maybe the update isn't complete yet.

Reply:
The new gymnasium env.step() function expects 5 values to be returned, while the old gymnasium package used to expect only 4 values

Reply:
by simply using "env_make_kwargs={...}" & replacing the pinned reference to imitation in "pyproject.toml" it was possible to get rid of this workaround